packages/x-date-pickers/src/DateCalendar/tests/localization.DateCalendar.test.tsx

```diff
@@ -44,9 +44,11 @@ describe('<DateCalendar /> - localization', () => {
             adapterLocale: adapterName === 'date-fns' ? fr : 'fr',
           });
 
-          expect(document.querySelector(`.${dayCalendarClasses.weekDayLabel}`)!.ariaLabel).to.equal(
-            'lundi',
-          );
+          await waitFor(() => {
```

Comment:
IMHO `waitFor` should be avoided if it is not necessary and only "hides" a flaky test problem. 🙈 
Especially given the findings in https://github.com/mui/mui-x/pull/16257